server/ERNI.PBA.Server.Host/Startup.cs

```diff
@@ -78,35 +80,28 @@ public void ConfigureServices(IServiceCollection services)
                         OnTokenValidated = async context =>
                         {
                             var db = context.HttpContext.RequestServices.GetRequiredService<DatabaseContext>();
-
                             var sub = context.Principal.Claims.Single(c => c.Type == "sub").Value;
-
                             var user = await db.Users.SingleOrDefaultAsync(_ => _.UniqueIdentifier == sub);
 
-                            if (user == null)
+                            var claims = new List<Claim>
                             {
-                                user = new User
-                                {
-                                    UniqueIdentifier = sub,
-                                    FirstName = context.Principal.Claims.Single(c => c.Type == "given_name").Value,
-                                    LastName = context.Principal.Claims.Single(c => c.Type == "family_name").Value,
-                                    Username = context.Principal.Claims.Single(c => c.Type == "upn").Value
-                                };
-                                db.Users.Add(user);
-                                db.SaveChanges();
-                            }
-
-                            var claims = new List<System.Security.Claims.Claim>();
-                            claims.Add(new System.Security.Claims.Claim(Claims.Id, user.Id.ToString()));
-                            claims.Add(new System.Security.Claims.Claim(Claims.FirstName, user.FirstName));
-                            claims.Add(new System.Security.Claims.Claim(Claims.LastName, user.LastName));
+                                new Claim(Claims.FirstName, context.Principal.Claims.Single(c => c.Type == "given_name").Value),
```

Comment:
these claims we already have, we do not need to create them again

Reply:
We do have them, but only in the token. Each time we want to use them, we need to write LINQ query. This stores them in statically typed class for easier use in the rest of the application.